### ISSUE TITLE
[Bug]: cue indexing recognized some of cue file，but some not.

### ISSUE THREAD

**Issue by Linxuanhua1** [kagemomiji/airsonic-advanced#624]
### What happened?

Not all of cue files are recognized correctly. The files encoded with utf8-bom aren't recognized. Does the programme not support this format?

### Steps to reproduce

Normally run docker version with enabling cue indexing.

### Version

11.1.5-SNAPSHOT (Edge)

### Operating System

Linux

### Java Version

docker included

### Database

Other

### Configuration paramter

Normal

### language

None

**Comment by yuygfgg** [comment]
could you please provide a sample cue file?

**Comment by Linxuanhua1** [comment]
[Uploading KSLA-0001.zip…]()

**Comment by Linxuanhua1** [comment]
> could you please provide a sample cue file?

[KSLA-0001.zip](https://github.com/user-attachments/files/17324410/KSLA-0001.zip)

**Comment by Linxuanhua1** [comment]
> could you please provide a sample cue file?

I found that some utf8-bom cue files were correctly recognized, but others were not. It's quite amazing

**Comment by UVJkiNTQ** [comment]
> > could you please provide a sample cue file?
> 
> I found that some utf8-bom cue files were correctly recognized, but others were not. It's quite amazing

I tried with the edge build and found all cues recogniztion are failed.

**Comment by UVJkiNTQ** [comment]
[airsonic.log](https://github.com/user-attachments/files/17787206/airsonic.log)
log attached

**Comment by UVJkiNTQ** [comment]
https://github.com/kagemomiji/airsonic-advanced/commit/cd6b82453278457f52b4ee73c8a10f069d0b3e88
This problem was introduced in this commit.
I tested former builds and that is okay.

**Comment by yuygfgg** [comment]
I'm quite confident that snapshots after https://github.com/kagemomiji/airsonic-advanced/commit/cd6b82453278457f52b4ee73c8a10f069d0b3e88 still works, at least those during May 2024. However, I'm not sure about later builds.

**Comment by yuygfgg** [comment]
@kagemomiji Could you please take a look? This seems to be an issue with wide-ranging implications.

**Comment by yuygfgg** [comment]
Only UTF-8 without BOM works

**Comment by kagemomiji** [comment]
@Linxuanhua1 @UVJkiNTQ @yuygfgg 
I have created fixed docker image. Could you try `ghcr.io/kagemomiji/airsonic-advanced:pr-636`?

**Comment by UVJkiNTQ** [comment]
> @Linxuanhua1 @UVJkiNTQ @yuygfgg I have created fixed docker image. Could you try `ghcr.io/kagemomiji/airsonic-advanced:pr-636`?

I will try

**Comment by Linxuanhua1** [comment]
> @Linxuanhua1 @UVJkiNTQ @yuygfgg I have created fixed docker image. Could you try `ghcr.io/kagemomiji/airsonic-advanced:pr-636`?

ok, I will try it now!!!!!

**Comment by UVJkiNTQ** [comment]
Seems okay for me in library recognization.
I will further test with palylist containing cue.

**Comment by Linxuanhua1** [comment]
awesome!! Thanks!!  It works!!

**Comment by UVJkiNTQ** [comment]
I further tried with m3u file containing cue path.

The file parsing seems okay but the 'extraction' of the play iitems is failed and I got a 'missing'  tag.
 [test.m3u.txt](https://github.com/user-attachments/files/17789009/test.m3u.txt)
![image](https://github.com/user-attachments/assets/3ea8f6e4-1699-4351-b5f3-50c79f0c13af)

Maybe this kind of palylist is not supported yet?

**Comment by kagemomiji** [comment]
@UVJkiNTQ 
No, Airsonic Advanced does not support M3U files that include CUE files.

> Thanks for the fix.
> 
> I further tried with m3u file containing cue path.
> 
> The file parsing seems okay but the 'extraction' of the play items is failed and I got a 'missing' tag. [test.m3u.txt](https://github.com/user-attachments/files/17789009/test.m3u.txt) ![image](https://private-user-images.githubusercontent.com/12570945/386922799-3ea8f6e4-1699-4351-b5f3-50c79f0c13af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MjI2MjMsIm5iZiI6MTczMTgyMjMyMywicGF0aCI6Ii8xMjU3MDk0NS8zODY5MjI3OTktM2VhOGY2ZTQtMTY5OS00MzUxLWI1ZjMtNTBjNzlmMGMxM2FmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE3VDA1NDUyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1NWZjNmExZWZjMjM1NTI4NTdjOGMxNDQ0YTdhYjMyNGZmYWQxYWM2NDZmZmMxMTIwZjA5NmY4ZmEwN2YwZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tAmw7dlurUSy0eOLvf3cW3sa50fmkR8litF9Dq7OHRI)
> 
> Maybe this kind of playlist is not supported yet?

**Comment by UVJkiNTQ** [comment]
Increased the library size and found more cues errors.
[KSLA-1009.cue.txt](https://github.com/user-attachments/files/17789901/KSLA-1009.cue.txt)

**Comment by UVJkiNTQ** [comment]
Further tested with 11.1.3-SNAPSHOT.20230429160252. It is tricky now.

**Comment by kagemomiji** [comment]
@UVJkiNTQ 
It seems that the CUE Sheets you cannot load have issues with their formatting, preventing them from being processed correctly.
I have implemented a feature to output a warning message whenever a CUE Sheet fails to load.
Please pull `pr-636` image again and check the warning message and adjust the CUE Sheet accordingly.

**Comment by UVJkiNTQ** [comment]
> @UVJkiNTQ 
> It seems that the CUE Sheets you cannot load have issues with their formatting, preventing them from being processed correctly.
> I have implemented a feature to output a warning message whenever a CUE Sheet fails to load.
> Please pull `pr-636` image again and check the warning message and adjust the CUE Sheet accordingly.

Thanks for your effort.
I will try it.
Btw, is it possible to have a seperate tool to do the validation just under the folder with the same parsing method? That might be more straightforward.

**Comment by UVJkiNTQ** [comment]
[CUEs.zip](https://github.com/user-attachments/files/17831984/CUEs.zip)
[airsonic.log](https://github.com/user-attachments/files/17831985/airsonic.log)
Tested with small sample.
@kagemomiji 
These are the results:

`The field is too long to burn as CD-TEXT. The maximum length is 80.`
This seems just a warning, parsing is okay.

`Wrong number of digits in number`
This seems error parsing. The time code is larger than 99 is first field.
`  TRACK 22 AUDIO
    TITLE "Highest Life"
    INDEX 01 108:48:69`

`Could not find base file 'Lia - Heartily Song.flac' for cue sheet [KSLM-0099]『Heartily Song』[flac+cue+log+png]/Lia - Heartily Song.flac`
This seems error parsing. The file is there in actual.

`Invalid position. First index must have position 00:00:00`
I have no idea about this. CUES attached.
This seems error parsing. 

`Unparseable line. Will ignore`
This is actual regain info. `REM REPLAYGAIN_TRACK_GAIN -10.4 dB`
This seems error parsing. 

`Empty lines not allowed. Will ignore`
Extra empty lines at the end of file. This is quite normal in early output of ripping tools like EAC,
This seems error parsing.

**Comment by kagemomiji** [comment]
@UVJkiNTQ 
Thank you for providing the information.

I plan to implement an extended functionality to parse CUE Sheets using this new data. 
For now, I will close this issue and address it in a new issue.
